[2024-09-09T16:06:24.331-0300] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-09T16:06:24.343-0300] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dag_save_database_raw.dag_save_database_raw manual__2024-09-09T19:04:58.078592+00:00 [queued]>
[2024-09-09T16:06:24.351-0300] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dag_save_database_raw.dag_save_database_raw manual__2024-09-09T19:04:58.078592+00:00 [queued]>
[2024-09-09T16:06:24.352-0300] {taskinstance.py:2865} INFO - Starting attempt 1 of 11
[2024-09-09T16:06:24.375-0300] {taskinstance.py:2888} INFO - Executing <Task(BashOperator): dag_save_database_raw> on 2024-09-09 19:04:58.078592+00:00
[2024-09-09T16:06:24.378-0300] {standard_task_runner.py:72} INFO - Started process 131306 to run task
[2024-09-09T16:06:24.380-0300] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'dag_save_database_raw', 'dag_save_database_raw', 'manual__2024-09-09T19:04:58.078592+00:00', '--job-id', '26', '--raw', '--subdir', 'DAGS_FOLDER/dag_save_database_raw.py', '--cfg-path', '/tmp/tmpfk2ont_z']
[2024-09-09T16:06:24.381-0300] {standard_task_runner.py:105} INFO - Job 26: Subtask dag_save_database_raw
[2024-09-09T16:06:24.433-0300] {task_command.py:467} INFO - Running <TaskInstance: dag_save_database_raw.dag_save_database_raw manual__2024-09-09T19:04:58.078592+00:00 [running]> on host DESKTOP-0CT35O1.
[2024-09-09T16:06:24.508-0300] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Matheus' AIRFLOW_CTX_DAG_ID='dag_save_database_raw' AIRFLOW_CTX_TASK_ID='dag_save_database_raw' AIRFLOW_CTX_EXECUTION_DATE='2024-09-09T19:04:58.078592+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-09-09T19:04:58.078592+00:00'
[2024-09-09T16:06:24.510-0300] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-09-09T16:06:24.511-0300] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-09-09T16:06:24.512-0300] {logging_mixin.py:190} INFO - Current task name:dag_save_database_raw state:running start_date:2024-09-09 19:06:24.344272+00:00
[2024-09-09T16:06:24.513-0300] {logging_mixin.py:190} INFO - Dag name:dag_save_database_raw and current dag run status:running
[2024-09-09T16:06:24.514-0300] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-09T16:06:24.516-0300] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-09-09T16:06:24.518-0300] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', "python '/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/save_database.py'"]
[2024-09-09T16:06:24.522-0300] {subprocess.py:86} INFO - Output:
[2024-09-09T16:06:29.291-0300] {subprocess.py:93} INFO - 24/09/09 16:06:29 WARN Utils: Your hostname, DESKTOP-0CT35O1 resolves to a loopback address: 127.0.1.1; using 10.255.255.254 instead (on interface lo)
[2024-09-09T16:06:29.294-0300] {subprocess.py:93} INFO - 24/09/09 16:06:29 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2024-09-09T16:06:29.793-0300] {subprocess.py:93} INFO - 24/09/09 16:06:29 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-09-09T16:06:29.956-0300] {subprocess.py:93} INFO - Setting default log level to "WARN".
[2024-09-09T16:06:29.957-0300] {subprocess.py:93} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-09-09T16:06:37.136-0300] {subprocess.py:93} INFO - [Stage 1:>                                                          (0 + 1) / 1]                                                                                24/09/09 16:06:37 WARN SparkStringUtils: Truncated the string representation of a plan since it was too large. This behavior can be adjusted by setting 'spark.sql.debug.maxToStringFields'.
[2024-09-09T16:06:37.474-0300] {subprocess.py:93} INFO - 24/09/09 16:06:37 ERROR Executor: Exception in task 0.0 in stage 2.0 (TID 2)
[2024-09-09T16:06:37.476-0300] {subprocess.py:93} INFO - org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:06:37.477-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:06:37.478-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:06:37.479-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.481-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:06:37.482-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:06:37.483-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:06:37.484-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:06:37.486-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:06:37.489-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.490-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.491-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:06:37.493-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:06:37.494-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:06:37.496-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:06:37.497-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:06:37.498-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:06:37.500-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:06:37.501-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:06:37.503-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:06:37.504-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:06:37.505-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:06:37.506-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:06:37.508-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:06:37.509-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:06:37.510-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:06:37.511-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:06:37.512-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:06:37.513-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:06:37.514-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:06:37.516-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:06:37.517-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:06:37.518-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:06:37.520-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:06:37.521-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:06:37.522-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:06:37.523-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:06:37.524-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:06:37.525-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.527-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:06:37.528-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:06:37.529-0300] {subprocess.py:93} INFO - 24/09/09 16:06:37 WARN TaskSetManager: Lost task 0.0 in stage 2.0 (TID 2) (10.255.255.254 executor driver): org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:06:37.530-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:06:37.531-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:06:37.532-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.533-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:06:37.534-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:06:37.535-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:06:37.536-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:06:37.537-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:06:37.539-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.539-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.540-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:06:37.542-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:06:37.543-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:06:37.544-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:06:37.545-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:06:37.546-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:06:37.547-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:06:37.548-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:06:37.549-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:06:37.550-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:06:37.551-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:06:37.552-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:06:37.553-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:06:37.554-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:06:37.555-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:06:37.557-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:06:37.558-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:06:37.559-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:06:37.560-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:06:37.561-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:06:37.562-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:06:37.563-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:06:37.564-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:06:37.565-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:06:37.566-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:06:37.567-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:06:37.568-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:06:37.569-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.570-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:06:37.571-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:06:37.572-0300] {subprocess.py:93} INFO - 
[2024-09-09T16:06:37.573-0300] {subprocess.py:93} INFO - 24/09/09 16:06:37 ERROR TaskSetManager: Task 0 in stage 2.0 failed 1 times; aborting job
[2024-09-09T16:06:37.604-0300] {subprocess.py:93} INFO - Traceback (most recent call last):
[2024-09-09T16:06:37.606-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/save_database.py", line 65, in <module>
[2024-09-09T16:06:37.607-0300] {subprocess.py:93} INFO -     dfempregados.write.jdbc(url, "empregados", mode="overwrite", properties=myproperties)
[2024-09-09T16:06:37.608-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/pyspark/sql/readwriter.py", line 1984, in jdbc
[2024-09-09T16:06:37.609-0300] {subprocess.py:93} INFO -     self.mode(mode)._jwrite.jdbc(url, table, jprop)
[2024-09-09T16:06:37.610-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
[2024-09-09T16:06:37.611-0300] {subprocess.py:93} INFO -     return_value = get_return_value(
[2024-09-09T16:06:37.612-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-09-09T16:06:37.613-0300] {subprocess.py:93} INFO -     return f(*a, **kw)
[2024-09-09T16:06:37.614-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/py4j/protocol.py", line 326, in get_return_value
[2024-09-09T16:06:37.616-0300] {subprocess.py:93} INFO -     raise Py4JJavaError(
[2024-09-09T16:06:37.663-0300] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o55.jdbc.
[2024-09-09T16:06:37.665-0300] {subprocess.py:93} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 2.0 failed 1 times, most recent failure: Lost task 0.0 in stage 2.0 (TID 2) (10.255.255.254 executor driver): org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:06:37.666-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:06:37.667-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:06:37.668-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.669-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:06:37.670-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:06:37.671-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:06:37.672-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:06:37.673-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:06:37.674-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.675-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.677-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:06:37.678-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:06:37.679-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:06:37.680-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:06:37.681-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:06:37.682-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:06:37.683-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:06:37.684-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:06:37.685-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:06:37.686-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:06:37.688-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:06:37.689-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:06:37.691-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:06:37.692-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:06:37.693-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:06:37.694-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:06:37.695-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:06:37.696-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:06:37.697-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:06:37.698-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:06:37.699-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:06:37.701-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:06:37.702-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:06:37.704-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:06:37.705-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:06:37.707-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:06:37.708-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:06:37.709-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.710-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:06:37.711-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:06:37.712-0300] {subprocess.py:93} INFO - 
[2024-09-09T16:06:37.713-0300] {subprocess.py:93} INFO - Driver stacktrace:
[2024-09-09T16:06:37.714-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-09-09T16:06:37.716-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-09-09T16:06:37.717-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-09-09T16:06:37.718-0300] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-09-09T16:06:37.720-0300] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-09-09T16:06:37.721-0300] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-09-09T16:06:37.722-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-09-09T16:06:37.723-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-09-09T16:06:37.724-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-09-09T16:06:37.725-0300] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2024-09-09T16:06:37.727-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-09-09T16:06:37.728-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-09-09T16:06:37.729-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-09-09T16:06:37.730-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-09-09T16:06:37.731-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-09-09T16:06:37.732-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-09-09T16:06:37.733-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-09-09T16:06:37.734-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-09-09T16:06:37.735-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-09-09T16:06:37.736-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2458)
[2024-09-09T16:06:37.737-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$1(RDD.scala:1039)
[2024-09-09T16:06:37.738-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
[2024-09-09T16:06:37.739-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
[2024-09-09T16:06:37.740-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.withScope(RDD.scala:410)
[2024-09-09T16:06:37.741-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:1037)
[2024-09-09T16:06:37.742-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$foreachPartition$1(Dataset.scala:3516)
[2024-09-09T16:06:37.743-0300] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2024-09-09T16:06:37.744-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withNewRDDExecutionId$1(Dataset.scala:4311)
[2024-09-09T16:06:37.745-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-09-09T16:06:37.746-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-09-09T16:06:37.747-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-09-09T16:06:37.748-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-09-09T16:06:37.750-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-09-09T16:06:37.751-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.withNewRDDExecutionId(Dataset.scala:4309)
[2024-09-09T16:06:37.751-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.foreachPartition(Dataset.scala:3516)
[2024-09-09T16:06:37.752-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.saveTable(JdbcUtils.scala:903)
[2024-09-09T16:06:37.753-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcRelationProvider.createRelation(JdbcRelationProvider.scala:65)
[2024-09-09T16:06:37.754-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:48)
[2024-09-09T16:06:37.756-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:75)
[2024-09-09T16:06:37.757-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:73)
[2024-09-09T16:06:37.758-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.command.ExecutedCommandExec.executeCollect(commands.scala:84)
[2024-09-09T16:06:37.759-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:107)
[2024-09-09T16:06:37.760-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-09-09T16:06:37.761-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-09-09T16:06:37.762-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-09-09T16:06:37.763-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-09-09T16:06:37.764-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-09-09T16:06:37.765-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:107)
[2024-09-09T16:06:37.766-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:98)
[2024-09-09T16:06:37.767-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:461)
[2024-09-09T16:06:37.768-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(origin.scala:76)
[2024-09-09T16:06:37.769-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:461)
[2024-09-09T16:06:37.770-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning(LogicalPlan.scala:32)
[2024-09-09T16:06:37.771-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning(AnalysisHelper.scala:267)
[2024-09-09T16:06:37.772-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning$(AnalysisHelper.scala:263)
[2024-09-09T16:06:37.773-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:32)
[2024-09-09T16:06:37.774-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:32)
[2024-09-09T16:06:37.775-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:437)
[2024-09-09T16:06:37.776-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.eagerlyExecuteCommands(QueryExecution.scala:98)
[2024-09-09T16:06:37.777-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.commandExecuted$lzycompute(QueryExecution.scala:85)
[2024-09-09T16:06:37.778-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.commandExecuted(QueryExecution.scala:83)
[2024-09-09T16:06:37.779-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.assertCommandExecuted(QueryExecution.scala:142)
[2024-09-09T16:06:37.780-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:859)
[2024-09-09T16:06:37.781-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:388)
[2024-09-09T16:06:37.782-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.saveInternal(DataFrameWriter.scala:361)
[2024-09-09T16:06:37.783-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:248)
[2024-09-09T16:06:37.784-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.jdbc(DataFrameWriter.scala:756)
[2024-09-09T16:06:37.785-0300] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-09-09T16:06:37.786-0300] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2024-09-09T16:06:37.787-0300] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-09-09T16:06:37.788-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2024-09-09T16:06:37.789-0300] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-09-09T16:06:37.790-0300] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-09-09T16:06:37.791-0300] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2024-09-09T16:06:37.792-0300] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-09-09T16:06:37.793-0300] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-09-09T16:06:37.794-0300] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-09-09T16:06:37.795-0300] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-09-09T16:06:37.796-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:06:37.797-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:06:37.798-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:06:37.799-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:06:37.800-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.801-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:06:37.802-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:06:37.803-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:06:37.804-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:06:37.805-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:06:37.806-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.807-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:06:37.808-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:06:37.809-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:06:37.811-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:06:37.812-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:06:37.812-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:06:37.813-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:06:37.814-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:06:37.815-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:06:37.817-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:06:37.817-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:06:37.818-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:06:37.819-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:06:37.820-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:06:37.821-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:06:37.822-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:06:37.823-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:06:37.824-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:06:37.825-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:06:37.826-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:06:37.827-0300] {subprocess.py:93} INFO - 	... 1 more
[2024-09-09T16:06:37.828-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:06:37.829-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:06:37.830-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:06:37.831-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:06:37.832-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:06:37.833-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:06:37.835-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:06:37.836-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:06:37.837-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:06:37.838-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:06:37.839-0300] {subprocess.py:93} INFO - 
[2024-09-09T16:06:38.115-0300] {subprocess.py:97} INFO - Command exited with return code 1
[2024-09-09T16:06:38.125-0300] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/operators/bash.py", line 249, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2024-09-09T16:06:38.144-0300] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-09-09T16:06:38.145-0300] {logging_mixin.py:190} INFO - Task start:2024-09-09 19:06:24.344272+00:00 end:2024-09-09 19:06:38.143959+00:00 duration:13.799687
[2024-09-09T16:06:38.146-0300] {logging_mixin.py:190} INFO - Task:<Task(BashOperator): dag_save_database_raw> dag:<DAG: dag_save_database_raw> dagrun:<DagRun dag_save_database_raw @ 2024-09-09 19:04:58.078592+00:00: manual__2024-09-09T19:04:58.078592+00:00, state:running, queued_at: 2024-09-09 19:04:58.124675+00:00. externally triggered: True>
[2024-09-09T16:06:38.147-0300] {logging_mixin.py:190} INFO - Failure caused by Bash command failed. The command returned a non-zero exit code 1.
[2024-09-09T16:06:38.148-0300] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=dag_save_database_raw, task_id=dag_save_database_raw, run_id=manual__2024-09-09T19:04:58.078592+00:00, execution_date=20240909T190458, start_date=20240909T190624, end_date=20240909T190638
[2024-09-09T16:06:38.175-0300] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-09T16:06:38.177-0300] {standard_task_runner.py:124} ERROR - Failed to execute job 26 for task dag_save_database_raw (Bash command failed. The command returned a non-zero exit code 1.; 131306)
Traceback (most recent call last):
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/operators/bash.py", line 249, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2024-09-09T16:06:38.228-0300] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-09T16:06:38.239-0300] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-09T16:06:38.241-0300] {local_task_job_runner.py:245} INFO - ::endgroup::
